catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/forms/SearchFormsApplicationTest.java

```diff
@@ -0,0 +1,438 @@
+      getContentsOfFile("form-simple.json");
+
+  private static final String TEMPLATE_FORM_METACARD_JSON_SIMPLE_RESPONSE =
+      removePrettyPrintingOnJson(getContentsOfFile("form-simple-response.json"));
+
+  private static final String TEMPLATE_FORM_FILTER_XML_SIMPLE =
+      removePrettyPrintingOnXml(getContentsOfFile("form-filter-simple.xml"));
+
+  private static final String CANNED_TITLE = "MY_TITLE";
+
+  private static final String CANNED_DESCRIPTION = "MY_DESCRIPTION";
+
+  private static final String CANNED_ID = "abcdefg";
+
+  private static final String CANNED_ISO_DATE = "2018-12-10T13:09:40Z";
+
+  private static final FilterBuilder MOCK_BUILDER = new GeotoolsFilterBuilder();
+
+  private static final CatalogFramework MOCK_FRAMEWORK = mock(CatalogFramework.class);
+
+  private static final Subject MOCK_SUBJECT = mock(Subject.class);
+
+  private static final ConfigurationApplication MOCK_CONFIG = mock(ConfigurationApplication.class);
+
+  private static final TemplateTransformer TRANSFORMER =
+      new TemplateTransformer(getWriter(), getRegistry());
+
+  private static final EndpointUtil UTIL =
+      new EndpointUtil(
+          null, // No interaction
+          MOCK_FRAMEWORK,
+          MOCK_BUILDER,
+          null, // No interaction
+          null, // No interaction
+          null, // No interaction
+          null, // No interaction
+          MOCK_CONFIG);
+
+  private static final SearchFormsApplication APPLICATION =
+      new SearchFormsApplication(
+          MOCK_FRAMEWORK, MOCK_BUILDER, TRANSFORMER, UTIL, () -> MOCK_SUBJECT);
+
+  // Will be initialized by setUpClass() when the port is known
+  private static String localhostFormsUrl = null;
+
+  private final ArgumentCaptor<CreateRequest> requestCaptor;
+
+  private final String formFilterXml;
+
+  private final String formRequestJson;
+
+  private final String formResponseJson;
+
+  // Ctor necessary for parameterization
+  public SearchFormsApplicationTest(String symbolUnderTest) {
+    this.requestCaptor = ArgumentCaptor.forClass(CreateRequest.class);
+    StrSubstitutor substitutor =
+        new StrSubstitutor(ImmutableMap.of("value", "hello" + symbolUnderTest));
+
+    this.formRequestJson = substitutor.replace(TEMPLATE_FORM_METACARD_JSON_SIMPLE);
+    this.formResponseJson = substitutor.replace(TEMPLATE_FORM_METACARD_JSON_SIMPLE_RESPONSE);
+
+    final String xmlVariation = JSON_TO_XML_SYMBOL_MAPPING.get(symbolUnderTest);
+    if (xmlVariation == null) {
+      this.formFilterXml = substitutor.replace(TEMPLATE_FORM_FILTER_XML_SIMPLE);
+    } else {
+      StrSubstitutor xmlSubstitutor =
+          new StrSubstitutor(ImmutableMap.of("value", "hello" + xmlVariation));
+      this.formFilterXml = xmlSubstitutor.replace(TEMPLATE_FORM_FILTER_XML_SIMPLE);
+    }
+  }
+
+  @BeforeClass
+  public static void setUpClass() {
+    // Scan from Spark's default port to some arbitrarily higher port
+    Spark.port(getFirstAvailablePortBetween(4567, 9999));
```

Comment:
Have we considered making this a component test since we are initializing spark?

Reply:
Yes. Looking into moving this test to Failsafe instead of Surefire.